src/components/CustomCheckbox.tsx

```diff
@@ -7,57 +7,70 @@ import {
   IndeterminateHighlightStar,
   NoHighlightStar,
 } from './Icons';
-import { CheckboxDisplayOptions } from '../types';
+import {
+  CheckboxDisplayOptionsHide,
+  CheckboxDisplayOptionsHighlight,
+  SelectionType,
+} from '../types';
 
 interface CustomCheckboxProps {
-  checkboxType: 'hide' | 'highlight';
+  selectionType: SelectionType;
   status: CheckboxState;
   clickHandler: () => void;
 }
 
 const CustomCheckbox: React.FunctionComponent<CustomCheckboxProps> = (
   props: CustomCheckboxProps
 ): JSX.Element => {
-  const { checkboxType, status, clickHandler } = props;
+  const { selectionType, status, clickHandler } = props;
 
-  const getDisplayOptions = (): CheckboxDisplayOptions => {
+  const getDisplayOptionsHighlight = (): CheckboxDisplayOptionsHighlight => {
     switch (status) {
       case CheckboxState.Checked:
         return {
-          hideTooltipText: 'Hide',
-          highlightTooltipText: 'Highlight',
+          highlightTooltipText: 'Remove highlight',
           highlightAriaLabel: 'true',
           highlightIcon: HighlightStar,
         };
       case CheckboxState.Unchecked:
         return {
-          hideTooltipText: 'Show',
-          highlightTooltipText: 'Remove highlight',
+          highlightTooltipText: 'Highlight',
           highlightAriaLabel: 'false',
           highlightIcon: NoHighlightStar,
         };
       case CheckboxState.Indeterminate:
         return {
-          hideTooltipText: 'Show',
-          highlightTooltipText: 'Remove highlight',
+          highlightTooltipText: 'Highlight',
           highlightAriaLabel: 'mixed',
           highlightIcon: IndeterminateHighlightStar,
         };
     }
   };
 
+  const getDisplayOptionsHide = (): CheckboxDisplayOptionsHide => {
+    switch (status) {
+      case CheckboxState.Checked:
+        return {
+          hideTooltipText: 'Hide',
+        };
+      case CheckboxState.Unchecked:
+        return {
+          hideTooltipText: 'Show',
+        };
+      case CheckboxState.Indeterminate:
+        return {
+          hideTooltipText: 'Show',
+        };
+    }
+  };
```

Comment:
```suggestion
  const tooltipMap =  {
        [CheckboxState.Checked]:'Hide',
        [CheckboxState.Unchecked]:'Show',
        [CheckboxState.Indeterminate]: 'Show',
    }
    
```

then below you just need to do `tooltipMap(status)

Reply:
just a few less lines of code